app/models/imo_delivery_detail.rb

```diff
@@ -0,0 +1,32 @@
+class ImoDeliveryDetail < ApplicationRecord
+  has_one :communication, as: :detailable
+
+  enum result: {
+    error: "error",
+    no_imo_account: "no_imo_account",
+    not_subscribed: "not_subscribed",
+    read: "read",
+    sent: "sent"
+  }
+
+  after_create :update_authorization, if: proc { |detail| detail.result.in? ["not_subscribed", "no_imo_account"] }
```

Comment:
Would you mind moving this responsibility out of a model callback and make it more of a directly called behavior.  Maybe something that is called in the Communication method above.

Having key control flow behavior like this hidden away in a callback can make things hard to follow and reason about down the road as more behavior and code gets added.